x-pack/plugins/security/public/session/session_expired.ts

```diff
@@ -4,20 +4,28 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
+import {
+  AUTH_PROVIDER_HINT_QUERY_STRING_PARAMETER,
+  LOGOUT_REASON_QUERY_STRING_PARAMETER,
+  NEXT_URL_QUERY_STRING_PARAMETER,
+} from '../../common/constants';
+
 export interface ISessionExpired {
   logout(): void;
 }
 
 const getNextParameter = () => {
   const { location } = window;
   const next = encodeURIComponent(`${location.pathname}${location.search}${location.hash}`);
-  return `&next=${next}`;
+  return `&${NEXT_URL_QUERY_STRING_PARAMETER}=${next}`;
 };
 
 const getProviderParameter = (tenant: string) => {
   const key = `${tenant}/session_provider`;
   const providerName = sessionStorage.getItem(key);
-  return providerName ? `&provider=${encodeURIComponent(providerName)}` : '';
+  return providerName
+    ? `&${AUTH_PROVIDER_HINT_QUERY_STRING_PARAMETER}=${encodeURIComponent(providerName)}`
```

Comment:
**note** this change initially confused me. I thought this was going to end up automatically logging the user back in for providers that don't require user interaction (anonymous, pki, kerberos, etc). Once I traced through the code, I understood that this was instead being used to inform the logout routine about which provider should handle the logout.

I'm _ok_ with this change, but now we have two distinct uses for `auth_provider_hint`: One is meant for public consumption, and will become part of our public API, and this new case is more if an implementation detail that's unrelated to the public API.

Was consistency the motivation for this change, or was there another reason we changed the parameter name?

Reply:
> Was consistency the motivation for this change, or was there another reason we changed the parameter name?

Yeah, just wanted to use less query string parameters and decided to re-use this one since its name isn't strictly bound to its current purpose yet and is more or less relevant to both cases. I don't have a strong on this though, if it feels confusing to you (and I see where the confusion is coming from) I can switch back to `provider` (or something even more descriptive). What do you think?

Reply:
I have a slight preference for `provider` over `auth_provider_hint`, but I don't feel strongly enough to have you change it. Happy for you to merge as-is

Reply:
Good :+1: I'll reverted back to `provider`. If it confused you then it can confuse someone else who reads this code in the future too